### DISCUSSION TITLE
Intercom integration

### DISCUSSION THREAD

**Discussion by Dakarain** [JoDehli/PyLoxone#331] [General]
Hi. I apologize, but I am a complete beginner in HA. I managed to get PyLoxone running, but I have a problem with integrating the intercom. Is there something specific that needs to be set up?

**Comment by JoDehli** [comment]
@Dakarain not all from the intercom is supported. Only some switches. But I do not have and use one.

**Comment by CrisBCros** [comment]
Hi JoDehli

First of all, a huge thanks for your work on PyLoxone! I recently set up my Home Assistant system with your integration, and I’m really grateful that it works so well with my Loxone setup.
I have a tablet running Home Assistant 24/7, displaying a dedicated dashboard. I was able to trigger an automation in Home Assistant by creating a virtual switch of my intercom in my Loxone config.

I have a couple of questions regarding video integration with my intercom, and I was hoping you could provide some insights:

1. Opening the Video Stream in an Automation

- I wanted to automatically open the intercom’s video stream fullscreen on the tablet when someone rings the doorbell.

3. Capturing an Image or a Short Clip (with Audio) in an Automation

- Ideally, I’d like to capture a picture or even a short video clip when the doorbell is pressed.
- If possible, including audio would be great.

Do you know if PyLoxone can be used for this, or would I need to integrate this separately with another Home Assistant component?

I’d really appreciate any guidance you can provide on this! Thanks again for all your work on this great project. 😊

Best regards,
Chris

**Reply by JoDehli** [reply]
@CrisBCros sorry but I do not have an intercom but I think the camera can be connected with WebRTC. I think pyloxone is not the right place to get the video. What I would do I would trY to find out what signal is emitted when someone rings. Maybe you can listen to the event with pyloxone and start an automation which starts a WebRTC connection. If this is successful you can pass it to a google device or Alexa device. But this what I would do if I had one.

**Comment by OlofBailey** [comment]
Has anyone had any luck integrating the Loxone intercom with home assistant?

**Reply by Bonsy-cmyk** [reply]
more or less yes,
still picture -> generic camera, use your login credentials
Video stream -> go2rtc (the link for stream is: http://your.intercom.ip/mjpg/video.mjpg) Port 80
Audio stream uses iptel.
Where I was stuck until my intercom decided to quit working was the bell, I still don't know how to get a binary sensor for the bell

**Reply by JoDehli** [reply]
@Bonsy-cmyk you can watch for the loxone events in home assistant an ring the bell. It the event I exposed via the websocket connection you can write down the uuid. But I think a lot of the intercom events are not exposed thru the websocket connection

**Reply by Bonsy-cmyk** [reply]
I already tried to set a marker in Loxone config an connect it to a light entity but no luck. Anyway since my intercom's videostream is out of order I'll just wait to switch to a new intercom, and it won't be a loxone one, thats for sure.
